src/platform/macos.rs

```diff
@@ -1,12 +1,17 @@
 extern crate libc;
+extern crate libloading;
 
 use std::os::unix::ffi::OsStrExt;
 use std::path::Path;
 use std::{io, ffi};
+use self::libloading::os::unix::{Library, Symbol};
 
-extern "stdcall" {
-	fn renamex_np(oldpath: *const libc::c_char, newpath: *const libc::c_char, flags: libc::c_uint) -> libc::c_int;
+unsafe fn get_renamex_np() -> Option<Symbol<unsafe extern fn (oldpath: *const libc::c_char, newpath: *const libc::c_char, flags: libc::c_uint) -> libc::c_int>> {
+	let lib = Library::this();
+	lib.get(b"renamex_np").ok()
+}
 
+extern "stdcall" {
```

Comment:
wait, why "stdcall"? macOS uses system-v as a default calling convention.

Reply:
Indeed, shouldn't this be failing? "stdcall" is only used on windows right?

Reply:
> "stdcall" is only used on windows right?

Yeah, that was my impression too.

Reply:
ah, I missed your comment. This is not the change introduced in this pr. I probably just copy-pasted `extern xxx` from windows files. Yet, it's still interesting why it works